### PR TITLE
Add Linux Docker image

### PR DESCRIPTION
**Pull request by gabrielbosio**
- Add make commands to build image and run container.
- Add linker flags necessary to make it work inside the container.

It was tested on an Apple M1 by running `make`, `make test` and `make SANITIZER_FLAGS=-fno-omit-frame-pointer valgrind` inside the container. The image size is 1.79 GB.

### CODE REVIEW

**Review by fmoletta — APPROVED** [review]
LGTM

**Comment by jrchatruc** [comment]
Works great! Just add a section to the README titled something like "Valgrind/Asan on macos" explaining all this so others are aware

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM rust:latest
+
+RUN apt update && \
+    apt install valgrind -y
+
+WORKDIR /usr/cairo-vm_in_C
+COPY . .
```
Comment (jrchatruc) [inline Dockerfile]: Is this `COPY . .` necessary if we will mount a volume when running it anyway?

**Review by fmoletta — COMMENTED** [review]
Maybe we should add a CI workflow that makes sure that it works (as changes in the project's dependencies may break it)

**Review by jrchatruc — APPROVED** [review]

**Review by mmsc2 — APPROVED** [review]

**Pull request merged**